src/rgw/rgw_usage_perf.h

```diff
@@ -0,0 +1,140 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <shared_mutex>
+#include <thread>
+#include <atomic>
+#include <chrono>
+
+#include "common/perf_counters.h"
+#include "rgw_usage_cache.h"
+#include "common/dout.h"
+
+// To avoid heavy header rgw_sal.h , we are adding a forward declaration here
+namespace rgw::sal {
+  class Driver;
+  class Bucket;
+  class User;
+}
+
+namespace rgw {
+
+// Performance counter indices
+enum {
+  l_rgw_usage_first = 920000,
+  l_rgw_usage_cache_hit,
+  l_rgw_usage_cache_miss,
+  l_rgw_usage_cache_update,
+  l_rgw_usage_cache_evict,
+  l_rgw_usage_last
+};
+
+class UsagePerfDoutPrefix : public DoutPrefixProvider {
+  CephContext* cct;
+public:
+  explicit UsagePerfDoutPrefix(CephContext* _cct) : cct(_cct) {}
+  
+  CephContext* get_cct() const override { return cct; }
+  unsigned get_subsys() const override { return ceph_subsys_rgw; }
+  std::ostream& gen_prefix(std::ostream& out) const override {
+    return out << "usage_perf: ";
+  }
+};
+
+class UsagePerfCounters {
+private:
+  CephContext* cct;
+  rgw::sal::Driver* driver;
+  std::unique_ptr<UsageCache> cache;
+  
+  mutable std::shared_mutex counters_mutex;
+  
+  // Track raw pointers for proper cleanup
+  std::unordered_map<std::string, PerfCounters*> user_perf_counters;
+  std::unordered_map<std::string, PerfCounters*> bucket_perf_counters;
+  
+  PerfCounters* global_counters;
+  
+  // Track active buckets and users for background refresh
+  std::unordered_set<std::string> active_buckets;
+  std::unordered_set<std::string> active_users;
+  mutable std::mutex activity_mutex;
+  
+  std::thread refresh_thread;
+  std::atomic<bool> shutdown_flag{false};
+  std::chrono::seconds refresh_interval{1200};
+  
+  void create_global_counters();
+  PerfCounters* create_user_counters(const std::string& user_id);
+  PerfCounters* create_bucket_counters(const std::string& bucket_name);
```

Comment:
i don't think we need separate perf counters for users vs buckets. instead, we can just give the counters for each bucket a prometheus label for the bucket user and tenant name. then when someone wants to see the usage for a given user in prometheus, they can sum over all buckets with that user/tenant label

used with the PerfCountersCache, that could look something like:
```c++
std::string key = ceph::perf_counters::key_create("rgw_bucket_usage", {
    {"tenant", tenant},
    {"user", user_id},
    {"bucket", bucket_name},
  });
auto counters = counters_cache.get(key);
counters->set(l_rgw_bucket_bytes, stats.bytes_used);
counters->set(l_rgw_bucket_objects, stats.num_objects);
```

Reply:
regarding the "user" and "account" labels, maybe using the name "owner" for both cases would make dashboard integration easier